install-script/install.sh

```diff
@@ -1315,140 +1281,39 @@ function write_remote_config_launchd() {
-
-    local ext_indentation
-    readonly ext_indentation="${2}"
-
-    if grep "ephemeral:" "${file}" > /dev/null; then
-        sed -i.bak -e "1,/ephemeral:/ s/ephemeral:.*$/ephemeral: true/" "${file}"
-    else
-        # write ephemeral: true on the top of sumologic: extension
-        sed -i.bak -e "1,/sumologic:/ s/sumologic:/sumologic:\\
-\\${ext_indentation}ephemeral: true/" "${file}"
-    fi
+    otelcol-config --enable-ephemeral
 }
 
 # write api_url to user configuration file
 function write_api_url() {
     local api_url
     readonly api_url="${1}"
 
-    local file
-    readonly file="${2}"
-
-    local ext_indentation
-    readonly ext_indentation="${3}"
-
-    # ToDo: ensure we override only sumologic `api_base_url`
-    if grep "api_base_url" "${file}" > /dev/null; then
-        sed -i.bak -e "s/api_base_url:.*$/api_base_url: $(escape_sed "${api_url}")/" "${file}"
-    else
-        # write api_url on the top of sumologic: extension
-        sed -i.bak -e "1,/sumologic:/ s/sumologic:/sumologic:\\
-\\${ext_indentation}api_base_url: $(escape_sed "${api_url}")/" "${file}"
-    fi
+    otelcol-config --set-api-url "$api_url"
 }
 
 # write opamp endpoint to user configuration file
 function write_opamp_endpoint() {
     local opamp_endpoint
     readonly opamp_endpoint="${1}"
 
-    local file
-    readonly file="${2}"
-
-    local ext_indentation
-    readonly ext_indentation="${3}"
-
-    # ToDo: ensure we override only sumologic `api_base_url`
-    if grep "endpoint" "${file}" > /dev/null; then
-        sed -i.bak -e "s/endpoint:.*$/endpoint: $(escape_sed "${opamp_endpoint}")/" "${file}"
-    else
-        # write endpoint on the top of sumologic: opamp: extension
-        sed -i.bak -e "1,/opamp:/ s/opamp:/opamp:\\
-\\${ext_indentation}endpoint: $(escape_sed "${opamp_endpoint}")/" "${file}"
-    fi
+    otelcol-config --set-opamp-endpoint "$opamp_endpoint"
 }
 
 # write tags to user configuration file
 function write_tags() {
     local fields
     readonly fields="${1}"
 
-    local file
-    readonly file="${2}"
-
-    local indentation
-    readonly indentation="${3}"
-
-    local ext_indentation
-    readonly ext_indentation="${4}"
-
-    local fields_indentation
-    readonly fields_indentation="${ext_indentation}${indentation}"
-
-    local fields_to_write
-    fields_to_write="$(echo "${fields}" | sed -e "s/^\\([^\\]\\)/${fields_indentation}\\1/")"
-    readonly fields_to_write
-
-    # ToDo: ensure we override only sumologic `collector_fields`
-    if grep "collector_fields" "${file}" > /dev/null; then
-        sed -i.bak -e "s/collector_fields:.*$/collector_fields: ${fields_to_write}/" "${file}"
-    else
-        # write installation token on the top of sumologic: extension
-        sed -i.bak -e "1,/sumologic:/ s/sumologic:/sumologic:\\
-\\${ext_indentation}collector_fields: ${fields_to_write}/" "${file}"
-    fi
+    for field in $fields
+    do
+        otelcol-config --add-tag "$field"
+    done
```

Comment:
--add-tag needs to look like `--add-tag k=v`, not sure if $field will look like that

Reply:
It does now, thanks to [this subtle change](https://github.com/SumoLogic/sumologic-otel-collector-packaging/pull/103/files#diff-d396e12fe6ca2006601d801b6bb14cda09dda8bf22fc69b921c7e6704074f896R344) :)